prereise/gather/demanddata/nrel_efs/tests/test_get_efs_data.py

```diff
@@ -1,107 +1,138 @@
 import os
+import zipfile
 
 import pandas as pd
 from pandas.testing import assert_frame_equal
 from powersimdata.network.usa_tamu.constants.zones import abv2state
 
 from prereise.gather.demanddata.nrel_efs.get_efs_data import (
+    _check_electrification_scenarios_for_download,
+    _check_path,
+    _check_technology_advancements_for_download,
+    _download_data,
+    _extract_data,
     account_for_leap_year,
-    download_demand_data,
-    download_flexibility_data,
     partition_demand_by_sector,
     partition_flexibility_by_sector,
 )
 
+# Create a dummy demand data set
+cont_states = sorted(set(abv2state) - {"AK", "HI"})
+dummy_demand_data = {
+    "Electrification": ["High"] * 4 * 48 * 8760,
+    "TechnologyAdvancement": ["Rapid"] * 4 * 48 * 8760,
+    "Year": [2030] * 4 * 48 * 8760,
+    "LocalHourID": sorted([i for i in range(1, 8761)] * 4 * 48),
+    "State": sorted([i for i in cont_states] * 4) * 8760,
+    "Sector": ["Commercial", "Industrial", "Residential", "Transportation"] * 48 * 8760,
+    "LoadMW": [1, 2, 3, 4] * 48 * 8760,
+}
+dummy_demand_df = pd.DataFrame(data=dummy_demand_data)
+dummy_demand_df.to_csv("EFSLoadProfile_High_Rapid.csv", index=False)
+
+# Create a .zip file of the dummy demand data set
+dummy_demand_df.to_csv("test_demand.csv", index=False)
+with zipfile.ZipFile("test.zip", "w") as z:
+    z.write("test_demand.csv")
+os.remove("test_demand.csv")
+
+# Create a dummy flexibility data set
+dummy_flex_data = {
+    "Electrification": ["High"] * 4 * 48 * 8760,
+    "TechnologyAdvancement": ["Rapid"] * 4 * 48 * 8760,
+    "Flexibility": ["Base"] * 4 * 48 * 8760,
+    "Year": [2030] * 4 * 48 * 8760,
+    "LocalHourID": sorted([i for i in range(1, 8761)] * 4 * 48),
+    "State": sorted([i for i in cont_states] * 4) * 8760,
+    "Sector": ["Commercial", "Industrial", "Residential", "Transportation"] * 48 * 8760,
+    "LoadMW": [1, 2, 3, 4] * 48 * 8760,
+}
+dummy_flex_df = pd.DataFrame(data=dummy_flex_data)
+dummy_flex_df.to_csv("EFSFlexLoadProfiles_High.csv", index=False)
```

Comment:
These lines should be encapsulated in a function and executed at test time, not import time.

Reply:
Okay, I'll make sure these are included in their respective functions.

Reply:
Done.